tests/test_tarfs.py

```diff
@@ -172,3 +172,44 @@ def test_getinfo(self):
         self.assertEqual(info.name, "foo")
         self.assertEqual(info.size, 0)
         self.assertIs(info.type, ResourceType.directory)
+
+
+class TestTarFSReadFromTarCFDotSlashName(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.tarfs = fs.archive.tarfs.TarReadFS(io.BytesIO(cls._build_fixture()))
+
+    @staticmethod
+    def _build_fixture():
+        with fs.memoryfs.MemoryFS() as tmpfs:
+            tmpfile = tmpfs.open("test.tar", "wb+")
+            with tarfile.open(mode="w", fileobj=tmpfile) as tf:
+                tf.addfile(tarfile.TarInfo("./file1"), io.StringIO())
+                info = tarfile.TarInfo("./sub")
+                info.type = tarfile.DIRTYPE
+                tf.addfile(info, io.BytesIO())
+                tf.addfile(tarfile.TarInfo("./sub/file2"), io.StringIO())
+            tmpfile.seek(0)
+            return tmpfile.read()
+
+    def test_listdir(self):
+        self.assertListEqual(sorted(self.tarfs.listdir('/')), ['file1', 'sub'])
+        self.assertListEqual(self.tarfs.listdir('/sub'), ['file2'])
+
+    def test_exists(self):
+        self.assertTrue(self.tarfs.exists('file1'))
+        self.assertTrue(self.tarfs.exists('sub/file2'))
+        self.assertTrue(self.tarfs.exists('/file1'))
+        self.assertTrue(self.tarfs.exists('/sub/file2'))
+        self.assertTrue(self.tarfs.exists('./file1'))
+        self.assertTrue(self.tarfs.exists('./sub/file2'))
+
+    def test_walker_info(self):
+        file1 = self.tarfs.getinfo('file1')
+        self.assertTrue(file1)
```

Comment:
I don't think you need the `assertTrue` here, since `getinfo` is always going to return an `Info` (and `bool(info)` is always true i think), or raise an exception.

Reply:
I meant that to mean truthy, really. But yeah it's redundant anyway.

Reply:
Fixed